src/Application/SaveMappings/SaveMappingsUseCase.php

```diff
@@ -18,11 +21,30 @@ class SaveMappingsUseCase {
 	public function __construct(
 		private SaveMappingsPresenter $presenter,
 		private MappingRepository $repository,
-		private array $allowedPredicates
+		private array $allowedPredicates,
+		private EntityIdParser $entityIdParser,
+		private MappingListSerializer $mappingListSerializer
 	) {
 	}
 
-	public function saveMappings( EntityId $entityId, MappingList $mappings ): void {
+	public function saveMappings( string $entityIdValue, string $mappingsJson ): void {
+		try {
+			$entityId = $this->entityIdParser->parse( $entityIdValue );
+		} catch ( EntityIdParsingException ) {
+			// TODO: $this->presenter->presentInvalidEntityId()
+			return;
+		}
+
+		// TODO: check if $entityId exists
+
+		try {
+			$mappings = $this->mappingListSerializer->fromJson( $mappingsJson );
+		} catch ( InvalidArgumentException ) {
+			// TODO: present invalid predicates where the Mapping constructor fails
+			return;
+		}
+
+		// TODO: "invalid" is too generic here. We have malformed predicates, disallowed predicates and invalid objects.
```

Comment:
Thoughs on where we draw the line in terms of validation feedback? Should we differentiate the validation issues in the response, or simply return the first thing that fails and simply say something is invalid?